hasher-matcher-actioner/terraform/main.tf

```diff
@@ -242,6 +242,154 @@ module "authentication" {
   webapp_and_api_shared_user_pool_client_id = var.webapp_and_api_shared_user_pool_client_id
 }
 
+/*
+ * # Submissions SQS:
+ * Submissions from the API are routed directly into a queue. Doing an SNS
+ * indirection **could** allow multiple lambdas to be listening for submissions,
+ * but, that would be costly because the lambda invocation would cost money.
+ *
+ * Instead, we will have a single hashing lambda capable of handling all
+ * content_types. If the content, because of its size or because of compute
+ * complexity can't be handled by this "base" lambda, it will be routed to
+ * another specially capable lambda queue.
+ *
+ * - This should soon absorb the pdq_images queue + SNS topic as the only queue
+ *   that we will publish submissions on.
+ *   If we have proven that the generic lambda can generate PDQ signals, we can 
+ *   do away with the PDQ specific infrastructure altogether.
+*/
+resource "aws_sqs_queue" "submissions_queue" {
+  name_prefix                = "${var.prefix}-submissions"
+  visibility_timeout_seconds = 300
+  message_retention_seconds  = 1209600
+  tags = merge(
+    var.additional_tags,
+    local.common_tags,
+    {
+      Name = "SubmissionsQueue"
+    }
+  )
+}
+
+
+/* Hashing Lambda Configuration Begins
+ *
+ * Pipe the submissions queue into the hashing lambda.
+ */
+resource "aws_lambda_function" "hashing_lambda" {
+  function_name = "${var.prefix}_hasher"
+  package_type  = "Image"
+  role          = aws_iam_role.hashing_lambda_role.arn
+  image_uri     = var.hma_lambda_docker_uri
+  image_config {
+    command = ["hmalib.lambdas.hashing.lambda_handler"]
+  }
+
+  timeout     = 300
+  memory_size = 512
+
+  environment {
+    variables = {
+      DYNAMODB_TABLE      = module.datastore.primary_datastore.name
+      METRICS_NAMESPACE   = var.metrics_namespace
+      MEASURE_PERFORMANCE = var.measure_performance ? "True" : "False"
+      # HASHES_QUEUE_URL    = module.pdq_signals.hashes_queue_url
```

Comment:
remove?

Reply:
Will be required once hashes are published.